app/routers/menstrual_predictor.py

```diff
@@ -0,0 +1,179 @@
+import datetime
+from typing import List
+
+from fastapi import APIRouter, Depends, Request, HTTPException
+from fastapi.responses import RedirectResponse
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_302_FOUND
+
+from app.database.models import Event, UserMenstrualPeriodLength
+from app.dependencies import get_db, templates
+from app.routers.event import create_event
+from app.internal.utils import create_model, get_current_user
+
+from loguru import logger
+
+
+router = APIRouter(
+    prefix="/menstrual_predictor",
+    tags=["menstrual_predictor"],
+    dependencies=[Depends(get_db)]
+)
+
+MENSTRUAL_PERIOD_CATEGORY_ID = 111
+
+
+@router.get("/")
+def join(request: Request, db: Session = Depends(get_db)):
+    current_user_id = get_current_user(db).id
```

Comment:
Maybe use the Depends mechanism?

Reply:
I did, tell me if that's what you meant. :)